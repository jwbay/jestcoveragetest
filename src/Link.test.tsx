import * as React from 'react';
import * as renderer from 'react-test-renderer';
const which = '';
const Link = require('./Link' + which).default;

for (var index = 0; index < 10; index++) {
 test('Link changes the class when hovered' + index, () => {
    const component = renderer.create(
      <Link page="http://www.facebook.com">Facebook</Link>
    );
    let tree = component.toJSON();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree.props.className).toBe('hovered');
  });
}
