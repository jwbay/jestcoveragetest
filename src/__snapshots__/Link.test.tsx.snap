exports[`test Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}>
  Facebook
</a>
`;

exports[`test Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}>
  Facebook
</a>
`;
